src/api/http.rs

```diff
@@ -198,10 +219,85 @@ fn nodes(bob: State<BobServer>) -> Json<Vec<Node>> {
     Json(nodes)
 }
 
+#[get("/disks/list")]
+fn disks_list(bob: State<BobServer>) -> Result<Json<Vec<DiskState>>, StatusExt> {
+    let backend = bob.grinder().backend().inner();
+    let (dcs, adc) = backend
+        .disk_controllers()
+        .ok_or_else(not_acceptable_backend)?;
+
+    let task = async move {
+        let mut disks = Vec::new();
+        for dc in dcs.iter().chain(std::iter::once(&adc)) {
+            let disk_path = dc.disk();
+            disks.push(DiskState {
+                name: disk_path.name().to_owned(),
+                path: disk_path.path().to_owned(),
+                is_active: dc.is_ready().await,
+            });
+        }
+        disks
+    };
+
+    let rt = Runtime::new().expect("create runtime");
+    let disks = rt.block_on(task);
+
+    Ok(Json(disks))
+}
+
 #[get("/metadata/distrfunc")]
 fn distribution_function(bob: State<BobServer>) -> Json<DistrFunc> {
     let mapper = bob.grinder().backend().mapper();
-    Json(DistrFunc { func: format!("{:?}", mapper.distribution_func()) })
+    Json(DistrFunc {
+        func: format!("{:?}", mapper.distribution_func()),
+    })
+}
+
+#[post("/disks/<disk_name>/start")]
+fn start_disk_controller(bob: State<BobServer>, disk_name: String) -> Result<StatusExt, StatusExt> {
+    use futures::stream::{FuturesUnordered, StreamExt};
+    let backend = bob.grinder().backend().inner();
+    let (dcs, adc) = backend
+        .disk_controllers()
+        .ok_or_else(not_acceptable_backend)?;
+    let target_dcs = dcs
+        .iter()
+        .chain(std::iter::once(&adc))
+        .filter(|dc| dc.disk_name() == disk_name)
+        .map(|dc| dc.run())
+        .collect::<FuturesUnordered<_>>();
+    if target_dcs.len() == 0 {
+        let err = format!("Disk Controller with name '{}' not found", disk_name);
+        warn!("{}", err);
+        return Err(StatusExt::new(Status::NotFound, false, err));
+    }
+    let rt = Runtime::new().expect("create runtime");
+    let task = async move {
+        let err_string = target_dcs
+            .fold(String::new(), |mut err_string, res| {
+                if let Err(e) = res {
+                    err_string.push_str(&(e.to_string() + "\n"));
+                }
+                async move { err_string }
+            })
+            .await;
+        if err_string.len() == 0 {
+            Ok(())
+        } else {
+            Err(err_string)
+        }
+    };
+    match rt.block_on(task) {
+        Ok(()) => {
+            let msg = format!(
+                "all disk controllers for disk '{}' successfully started",
+                disk_name
+            );
+            info!("{}", msg);
+            Ok(StatusExt::new(Status::Ok, true, msg))
+        }
+        Err(e) => Err(StatusExt::new(Status::Ok, false, e.to_string())),
```

Comment:
Why do you return Status::Ok in case of an error?

Reply:
There may be 2 kinds of errors:
1) The requested disk_name doesn't exist
2) The requested disk can't start
I decided to return Status::NotFound for the first kind of errors, and false on `ok` flag in `StatusExt` struct, when disk can't start, but I'll change it on Status::InternalServerError